test/Tests.Infrastructure/TestResourcesAnalyzer.cs

```diff
@@ -33,6 +35,21 @@ static TestResourcesAnalyzer()
             _timer = new Timer(ProcessQueue, null, TimeSpan.FromSeconds(15), TimeSpan.FromSeconds(15));
         }
 
+        private static string GetDisplayName(ITestMethod testMethod)
+            => $"{testMethod.TestClass.Class.Name}::{testMethod.Method.Name}()";
+
+        internal static void Start(ITestMethod testMethod)
+        {
+            if (_enabled)
+                _resources.Enqueue(TestResources.Create(GetDisplayName(testMethod), nameof(Start)));
+        }
+
+        internal static void End(ITestMethod testMethod)
+        {
+            if (_enabled)
+                _resources.Enqueue(TestResources.Create(GetDisplayName(testMethod), nameof(End)));
+        }
+
         internal static void Start(Context context)
```

Comment:
Need a cleanup here. We do not want to keep all the methods. Also we want to make it synchronous.

Reply:
I will refactor this to be sync

Reply:
Done (in the next PR)